### PR TITLE
Scope, Types, Exceptions Content

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: src/chapters/exceptions/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      :caption: Code
+
+      console.log("This is" an example);
+
+   .. sourcecode:: js
+      :caption: Output
+
+      SyntaxError: missing ) after argument list
+
+   We put our second quotation mark in incorrect place, and so JavaScript doesn't know what to do with the second half of our phrase, and so a SyntaxError is thrown with the message: "missing ) after argument list".
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: "We put our second quotation mark in the incorrect place. JavaScript does not know what to do with the second half of our phrase and throws a ``SyntaxError`` with the message: ``missing ) after argument list``."

File: src/chapters/exceptions/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+JavaScript has built in exceptions with pre-defined names, and messages, however JavaScript also gives you the ability to create your own error messages.
+
+You have undoutedly experienced various Exceptions already throughout this class. Let's look at a few common Exceptions.
+
+Common Exceptions
+-----------------
+
+JavaScript has some built in Exceptions you may have already encountered in this class.
+
+One of the most common errors in Javascript is a **SyntaxError** which is thrown when we include a symbol JavaScript is not expecting.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: ``SyntaxError``

File: src/chapters/exceptions/introduction.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+   .. sourcecode:: js
+      :caption: Output
+
+      SyntaxError: missing ) after argument list
+
+   We put our second quotation mark in incorrect place, and so JavaScript doesn't know what to do with the second half of our phrase, and so a SyntaxError is thrown with the message: "missing ) after argument list".
+
+
+A **ReferenceError** is thrown when we try to use a variable that has not yet been defined.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: ``ReferenceError``

File: src/chapters/exceptions/introduction.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      :caption: Code
+
+      console.log(x[0]);
+
+   .. sourcecode:: js
+      :caption: Output
+
+      ReferenceError: x is not defined
+
+   We attempt to print out the first element in the variable x, but we never declare x. JavaScript throws a ReferenceError with the message: "x is not defined".
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: "... we never declared x."

``ReferenceError``

Comment (gildedgardenia) [reply src/chapters/exceptions/introduction.rst]: ``x is not defined``

File: src/chapters/exceptions/introduction.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+A **TypeError** is thrown when JavaScript expects something to be one type, but the provided value is a different type.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: ``TypeError``

File: src/chapters/exceptions/introduction.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      const a = "Launch";
+      
+      a = "Code";
+
+   .. sourcecode:: js
+      :caption: Output
+
+      TypeError: invalid assignment to const `a'
+
+   In this case we declare a constant as the string "Launch", and then try to change the immutable variable to "Code". JavaScript throws a TypeError with the message: "invalid assignment to const `b`".
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: "In this case, ..."

``TypeError`` with the message: ``invalid assignment to const "b"``

File: src/chapters/exceptions/introduction.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      :caption: Code
+
+      const a = "Launch";
+      
+      a = "Code";
+
+   .. sourcecode:: js
+      :caption: Output
+
+      TypeError: invalid assignment to const `a'
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: 'a'

File: src/chapters/exceptions/introduction.rst [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+   In this case we declare a constant as the string "Launch", and then try to change the immutable variable to "Code". JavaScript throws a TypeError with the message: "invalid assignment to const `b`".
+
+Exceptions give us a way to provide more information on how something went wrong. JavaScript's built in Exceptions are regularly used in the debugging process.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: built-in

File: src/chapters/exceptions/introduction.rst [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+ReferenceError: x is not defined
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: What is this line doing here?

File: src/chapters/exceptions/introduction.rst [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+There are more built in Exceptions in Java, you can read more by referencing the `MDN Errors Documentation <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors>`_ or `W3Schools JavaScript Error <https://www.w3schools.com/js/js_errors.asp>`_ resource.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: built-in

Comment (gildedgardenia) [reply src/chapters/exceptions/introduction.rst]: "There are more built-in exceptions in JavaScript. You can read more in the ..."

Comment (gildedgardenia) [reply src/chapters/exceptions/introduction.rst]: Remove the word resource at the end.

File: src/chapters/exceptions/introduction.rst [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+In the next section we will learn how to raise our own exceptions using the throw statement.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/introduction.rst]: ``throw``

File: src/chapters/exceptions/throw.rst [hunk 12]
```diff
@@ -0,0 +1,64 @@
+Throw
+=====
+
+In most programming languages when the compiler, or interpreter encounters code it doesn't know how to handle it has to notify the programmer that something has gone wrong. We call this act "throwing an exception". This is where the term throw comes from.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: "In most programming languages, when the compiler or interpreter encounters code it doesn't know how to handle, it throws an exception. This is how the compiler notifies the programmer that something has gone wrong."

The last sentence is unnecessary.

File: src/chapters/exceptions/throw.rst [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+JavaScript gives us a throw statement to raise exceptions whenever we need to inform the programmer that something happened that the programming language doesn't know how to handle.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: Delete this sentence.

File: src/chapters/exceptions/throw.rst [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+JavaScript gives us the ability to raise exceptions using the **throw** statement.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: ``throw``

File: src/chapters/exceptions/throw.rst [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+Default Error
+-------------
+
+We can throw a default Error object by using the throw statement, and passing the message in as an argument.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: ``throw`` for the second instance. Comma is unnecessary.

File: src/chapters/exceptions/throw.rst [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+      :caption: Code
+
+      throw ("You cannot divide by zero!");
+
+   .. sourcecode:: js
+      :caption: Output
+
+      Error: You cannot divide by zero!
+
+   We can throw a default Error and define the message.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: Delete this sentence.

File: src/chapters/exceptions/throw.rst [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+      throw SyntaxError("That is the incorrect syntax");
+
+   .. sourcecode:: js
+      :caption: Output
+
+      SyntaxError: That is the incorrect syntax
+
+   We can throw built in Exceptions, and change the message that is displayed.
+
+JavaScript gives us flexibiltiy by allowing us to raise standard library errors, and to define our own errors. We can use exceptions to allow our program to break, and provide useful information as to why something went wrong.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: This sentence should replace the one on line 32.

Comment (gildedgardenia) [reply src/chapters/exceptions/throw.rst]: Commas are not necessary for either sentence.

File: src/chapters/exceptions/throw.rst [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+Custom Error
+------------
+
+JavaScript will also let you define new types of Errors, by creating a new Error, and defining it's name, and message, however created our own errors goes beyond the scope of this class.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/throw.rst]: "JavaScript will also let you define new types of Errors. You may find this helpful in the future, however, that goes beyond the scope of this class."

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 19]
```diff
@@ -0,0 +1,72 @@
+Exceptions-Control-Flow
+=======================
+
+In some instances we receive input from a user, and we can't always ensure they will type in something that will run perfectly with our code.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: "In some instances, we receive input from a user and we cannot always ensure they will type in something that will run with our code."

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+JavaScript gives us some control flow tools to handle exceptions, by catching thrown exceptions.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: Comma is unnecessary.

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
+When we are anticipating an exception, and we want our program to continue running, and not experience the runtime error we can tell JavaScript to **try** to run a block of code, and if a specific exception is thrown to **catch** the exception and to run a different block of code.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: This sentence is confusing.

"We may be anticipating an exception, but still want our program to continue running and not experience a runtime error. We can tell JavaScript to **try** to run a block of code and if an exception is thrown, to **catch** the exception and run a different block of code."

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+          console.log("0");
+      }
+
+   .. sourcecode:: js
+      :caption: Output
+
+      We caught a ReferneceError, but our program continues to run!
+      0
+
+   In this example we attept to print out the first element of our variable x. We forgot to define x, but nested our code inside of a *try* statement, and specifically **catch** ReferenceErrors. It should be noted that only ReferenceErrors will be caught by the catch statement. However, we can define as many catch statements as we want.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: In this example, we attempt to print ... variable ``x``."

try is not bolded in the second sentence. Also, the comma before "and" is unnecessary. 

``ReferenceError``

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+      }
+      catch(ReferenceError) {
+          console.log("We caught a ReferenceError, but our program continues to run!");
+          console.log("0");
+      }
+
+   .. sourcecode:: js
+      :caption: Output
+
+      We caught a ReferneceError, but our program continues to run!
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: ``ReferenceError``

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+      0
+
+   In this example we attept to print out the first element of our variable x. We forgot to define x, but nested our code inside of a *try* statement, and specifically **catch** ReferenceErrors. It should be noted that only ReferenceErrors will be caught by the catch statement. However, we can define as many catch statements as we want.
+
+JavaScript also provides us with a **finally** statement, a block of code that always runs every time regardless to if the try, or catch blocks run.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: ``finally``

"statement: a block of code ... regardless if the ``try`` or ``catch`` blocks run."

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+          console.log("0");
+      }
+      finally {
+          console.log("In the finally statement!");
+      }
+
+   .. sourcecode:: js
+      :caption: Output
+
+      We caught a ReferenceError, but our program continues to run!
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: ``ReferenceError``

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+      0
+      In the finally statement!
+
+   This try, catch, block is identical to the example above, except this time we have a finally statement. The finally statement runs upon the conclusion of the try, catch statements. It is often used to remove hanging resources.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: "try...catch block"

File: src/chapters/exceptions/exceptions-control-flow.rst [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+     
+Errors are a normal part of coding. JavaScript defines the most common errors, and provides messages to help us debug. We can extend the functionality around exceptions to further assist us in debugging, or to handle expected issues in our code.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exceptions-control-flow.rst]: Comma unnecessary in second and third sentences.

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 26 above)
Comment (gildedgardenia) [reply src/chapters/exceptions/exceptions-control-flow.rst]: ``finally`` 

try...catch

File: src/chapters/scope/introduction.rst [hunk 28]
```diff
@@ -0,0 +1,144 @@
+Introduction
+============
+
+.. index:: ! scope
+
+Where variables are declared and initialized in the code can effect whether or not they can be used.
+**Scope** refers to the ability to access and use the variable.
```
Comment (pdmxdd) [inline src/chapters/scope/introduction.rst]: Should it read: "**Scope** refers to the ability to access and use a variable." or "**Scope** refers to the ability to access and use the variable."?

File: src/chapters/scope/introduction.rst [hunk 29]
```diff
@@ -0,0 +1,144 @@
...
+
+	``a`` is accessible *inside* and *outside* of ``coolFunction()``. 
+	``b`` is only accessible *inside* of ``coolFunction()``.
+
+Block/Local Scope
+-----------------
+
+.. index:: ! local scope
+
+**Local scope** refers to variables declared and initialized inside a function.
```
Comment (pdmxdd) [inline src/chapters/scope/introduction.rst]: "**Local scope** refers to variables declared and initialized inside a function or block."

File: src/chapters/more-on-types/specialtypes.rst [hunk 30]
```diff
@@ -0,0 +1,111 @@
...
+
+While object types such as objects and arrays are mutable, primitive data types are immutable.
+Immutable data types are data types that cannot be changed once the value has been created.
+
+Primitive data types include:
+
+1. Strings
+2. Numbers
+3. Booleans
+4. ``null``
```
Comment (pdmxdd) [inline src/chapters/more-on-types/specialtypes.rst]: Switch the order of null and undefined, so it follows the writeup below.

File: src/chapters/exceptions/introduction.rst [hunk 31]
```diff
@@ -0,0 +1,97 @@
+Introduction
+============
+
+.. index:: ! scope
+
+Errors are a part of coding. Occasionally we make mistakes as programmers. When we make mistakes we look for information to figure out where we went wrong.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: Updated

File: src/chapters/exceptions/introduction.rst [hunk 32]
```diff
@@ -0,0 +1,97 @@
...
+
+Earlier in this course we learned about two different types of errors: Runtime and Logic. A logic error is when your program executes without breaking, but doesn't behave the way you thought it would. These logic errors usually require you to consider how you are going about solving the issue to resolve. Runtime errors are when your program does not run correctly, and an Exception is raised.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: updated

[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: updated

File: src/chapters/exceptions/introduction.rst [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+
+An **Exception** is a runtime error in which a name, and message are displayed to provide more information about the error.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: updated

[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: updated

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 23 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: Updated

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 24 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: I used double ` before finally, try, and catch. Is that correct? Also removed the unnecessary "to".

File: src/chapters/exceptions/exercises.rst [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+=========
+
+Zero Division: Throw
+--------------------
+
+Write a function called divide that takes two parameters: a numerator, and a denominator.
+
+Your function should return the result of numerator / denominator.
+
+However if the denominator is zero you should throw an error informing the user they attempted to divide by zero.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exercises.rst]: "However, ..."

File: src/chapters/exceptions/exercises.rst [hunk 35]
```diff
@@ -0,0 +1,47 @@
+Exercises
+=========
+
+Zero Division: Throw
+--------------------
+
+Write a function called divide that takes two parameters: a numerator, and a denominator.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exercises.rst]: "... a numerator and a denominator."

File: src/chapters/exceptions/exercises.rst [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+
+Your function should return the result of numerator / denominator.
+
+However if the denominator is zero you should throw an error informing the user they attempted to divide by zero.
+
+.. hint::
+
+   You should use an if, and throw statement to complete this exercise.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exercises.rst]: "... if and throw ..."

File: src/chapters/exceptions/exercises.rst [hunk 37]
```diff
@@ -0,0 +1,47 @@
...
+       return ((assignmentScore) / assignmentTotal);
+   }
+
+   let total = 25;
+
+   convertGrade(score, total);
+
+Upon running this code the teacher gets a ReferenceError: score is not defined. The teacher has hundreds of students and changing all of the missing grades to zeroes would take a long time.
+
+We can help by using a try statement, and a catch statement. Add a try, and catch statement for the ReferenceError and substitue a zero for any missing grades.
```
Comment (gildedgardenia) [inline src/chapters/exceptions/exercises.rst]: "try...catch"

File: src/chapters/exceptions/introduction.rst [hunk 38]
```diff
@@ -0,0 +1,97 @@
...
+
+Exceptions and Errors
+---------------------
+
+In JavaScript a runtime error, and an exception are the same thing, and can be used interchangeably. This can cause confusion because a logic error is not an exception!
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: removed

File: src/chapters/exceptions/introduction.rst [hunk 39]
```diff
@@ -0,0 +1,97 @@
...
+
+Error Object
+------------
+
+When a runtime error, also known as an execption, is raised JavaScript returns an Error object. An Error Object has two properties: a name, and a message. The name refers to the type of error that occured, while the message gives the user information on why that exception occured.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst [hunk 40]
```diff
@@ -0,0 +1,97 @@
...
+---------------------
+
+In JavaScript a runtime error, and an exception are the same thing, and can be used interchangeably. This can cause confusion because a logic error is not an exception!
+
+Error Object
+------------
+
+When a runtime error, also known as an execption, is raised JavaScript returns an Error object. An Error Object has two properties: a name, and a message. The name refers to the type of error that occured, while the message gives the user information on why that exception occured. 
+
+JavaScript has built in exceptions with pre-defined names, and messages, however JavaScript also gives you the ability to create your own error messages.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/scope/introduction.rst (hunk 28 above)
Comment (gildedgardenia) [reply src/chapters/scope/introduction.rst]: I like the first one!

File: src/chapters/exceptions/introduction.rst [hunk 41]
```diff
@@ -0,0 +1,97 @@
...
+
+You have undoutedly experienced various Exceptions already throughout this class. Let's look at a few common Exceptions.
+
+Common Exceptions
+-----------------
+
+JavaScript has some built in Exceptions you may have already encountered in this class.
```
[the remark this answers is not in the document]
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed both instances

File: src/chapters/exceptions/introduction.rst (hunk 2 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/scope/introduction.rst (hunk 29 above)
Comment (gildedgardenia) [reply src/chapters/scope/introduction.rst]: Thanks!

File: src/chapters/more-on-types/specialtypes.rst (hunk 30 above)
Comment (gildedgardenia) [reply src/chapters/more-on-types/specialtypes.rst]: Thanks!

File: src/chapters/exceptions/introduction.rst (hunk 1 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 3 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: resolved

File: src/chapters/exceptions/introduction.rst (hunk 4 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 5 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 6 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 7 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 8 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: all instances fixed

File: src/chapters/exceptions/introduction.rst (hunk 9 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: It was just accidentally copied in. It has been deleted.

File: src/chapters/exceptions/introduction.rst (hunk 10 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: fixed

File: src/chapters/exceptions/introduction.rst (hunk 11 above)
Comment (pdmxdd) [reply src/chapters/exceptions/introduction.rst]: resolved

File: src/chapters/exceptions/throw.rst (hunk 12 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: fixed

File: src/chapters/exceptions/throw.rst (hunk 13 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: deleted

File: src/chapters/exceptions/throw.rst (hunk 14 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: fixed

File: src/chapters/exceptions/throw.rst (hunk 15 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: fixed

File: src/chapters/exceptions/throw.rst (hunk 16 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: deleted

File: src/chapters/exceptions/throw.rst (hunk 17 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: fixed

File: src/chapters/exceptions/throw.rst (hunk 18 above)
Comment (pdmxdd) [reply src/chapters/exceptions/throw.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 19 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 20 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 21 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 22 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 24 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed after talking to Sally

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 25 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: This is already in a code output block. Leaving it as it is.

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 26 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exceptions-control-flow.rst (hunk 27 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exceptions-control-flow.rst]: fixed

File: src/chapters/exceptions/exercises.rst (hunk 34 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exercises.rst]: fixed

File: src/chapters/exceptions/exercises.rst (hunk 36 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exercises.rst]: fixed

File: src/chapters/exceptions/exercises.rst (hunk 37 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exercises.rst]: fixed

File: src/chapters/exceptions/exercises.rst (hunk 35 above)
Comment (pdmxdd) [reply src/chapters/exceptions/exercises.rst]: fixed

File: src/chapters/exceptions/studio.rst [hunk 42]
```diff
@@ -0,0 +1,23 @@
...
+--------
+
+Think of a bug you have seen in your code.
+This could be the time you dropped a keyword when initializing a variable or misused a method.
+
+1. Take some time to discuss with the group what your error was and how you solved it.
+   Did you talk to a TA to get it?
+   Did you find a great resource online that was helpful?
+
+Your TA will go over the pros and cons of different resources that can help you resolve the error.
```
Comment (pdmxdd) [inline src/chapters/exceptions/studio.rst]: I think we should provide them with a basic list of places to go when they encounter an error.

1. Google the error
2. Look for examples on Stackoverflow
3. Look over the documentation
4. Talk to a coworker, mentor, or TA
5. Describe the problem in detail. ie what information is coming in, and what is the desired behavior or output.

**Pull request merged**